File: JS/Math/GridUniquePaths.js
module.exports = {
	//param A : integer
	//param B : integer
	//return an integer
	uniquePaths : function(A, B){
        var nPc = function(n, c){
    	    var c1 = Math.min(c, n-c);
    	    var c2 = Math.max(c, n-c);
    	    var res = 1;
    	    for(var i = 1; i <= c1; i++){
    	        res *= (n - i + 1);
    	        res /= i;
    	    }
    	    return res;
        };

        return nPc(A + B - 2, A - 1);
	}
};

File: Java/Math/GridUniquePaths.java
public class Solution {
	public int uniquePaths(int a, int b) {
	    return nPc(a+b-2, a-1);
	}

	public int nPc(int n, int c){
	    int c1 = Math.min(c, n-c);
	    int c2 = Math.max(c, n-c);
	    long res = 1;
	    for(int i = c2+1; i <= n; i++)
	        res *= i;
	   for(int i = 2; i<= c1; i++)
	    res /= i;
	    return (int)res;
	}
}
